src/Controller/PageAdminController.php

```diff
@@ -0,0 +1,137 @@
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Security\Core\Exception\AccessDeniedException;
+
+/**
+ * Class PageAdminController.
+ *
+ * @author Vincent Laurier <vincent.laurier@ekino.com>
+ * @author Romain Mouillard <romain.mouillard@ekino.com>
+ * @author Ramesses Bonhof Keny <ramesses-bonhof.keny@ekino.com>
+ * @author Christian Kollross <christian.kollross@ekino.com>
+ */
+class PageAdminController extends BasePageAdminController
+{
+    /**
+     * @var BlockFilter
+     */
+    private $blockFilter;
+
+    /**
+     * PageAdminController constructor.
+     */
+    public function __construct(BlockFilter $blockFilter)
+    {
+        $this->blockFilter = $blockFilter;
+    }
+
+    /**
+     * @param Request|null $request
+     *
+     * @throws AccessDeniedException
+     * @throws NotFoundHttpException
+     *
+     * @return Response
+     */
+    public function composeContainerShowAction(Request $request = null)
+    {
+        /** @var BlockAdmin $blockAdmin */
+        $blockAdmin = $this->get('sonata.page.admin.block');
+
+        if (false === $blockAdmin->isGranted('LIST')) {
+            throw new AccessDeniedException();
+        }
+
+        $id    = $request->get($this->admin->getIdParameter());
+        $block = $blockAdmin->getObject($id);
+
+        if (!$block) {
+            throw new NotFoundHttpException(sprintf('Unable to find the block with id : %s', $id));
+        }
+
+        /** @var BlockServiceManager $blockServiceManager */
+        $blockServiceManager = $this->get('sonata.block.manager');
+        $blockServices       = $blockServiceManager->getServicesByContext('sonata_page_bundle', false);
+
+        // Filter service using the template configuration
+        /** @var Page $page */
+        if ($page = $block->getPage()) {
+            /** @var TemplateManagerInterface $templateManager */
+            $templateManager = $this->get('sonata.page.template_manager');
+            $template        = $templateManager->get($page->getTemplateCode());
+            $container       = $template->getContainer($block->getSetting('code'));
+
+            if (isset($container['blocks']) && \count($container['blocks']) > 0) {
+                foreach ($blockServices as $code => $service) {
+                    if (\in_array($code, $container['blocks'])) {
+                        continue;
+                    }
+
+                    unset($blockServices[$code]);
+                }
+            }
+
+            // We're filtering remaining block services according to page
+            $blockServices = $this->blockFilter->filter($blockServices, $page);
+        }
+
+        $blocksByCategory = $this->getBlocksByCategory($blockServices);
+
+        return $this->renderWithExtraParams('@SonataHelpers/PageAdmin/compose_container_show.html.twig', [
```

Comment:
can you add a todo to explain that it is done like this for now but we will have to use template registry to get the template name in the future plz?